---
title: "Label_Clusters"
output: html_document:
  fig_crop: no

date: "2024-05-01"
---
# Setup 
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(Seurat)
library(dplyr)
```

# Load Data
```{r}
data = readRDS("~/Documents/Kate/NMMscRNAseq/Data/integrated_merged_sct_filtered_tnbc_object.rds")

Idents(data) <- data@meta.data$SCT_snn_res.0.6
```


# Visualize Data
```{r}
DimPlot(data)
```

# Visualize Immune vs Stromal Populations
```{r}
FeaturePlot(data, features = c("Ptprc", "Col1a1", "Cd44"))
```
# Optimize Resolution Parameter
```{r}
data <- FindClusters(data, res = c(0.4, 0.6, 0.8, 1.0, 1.2, 1.4))
```
Look at different resolutions 
```{r}
# explore resolutions 
Idents(data) <- "SCT_snn_res.0.8"

DimPlot(data, reduction = "umap", label = T, label.size = 6)
```
```{r}
Idents(data) <- "SCT_snn_res.0.6"

DimPlot(data, reduction = "umap", label = T, label.size = 6)
```


```{r}
Idents(data) <- "SCT_snn_res.0.4"

DimPlot(data, reduction = "umap", label = T, label.size = 6)
```

```{r}
Idents(data) <- "SCT_snn_res.1.4"

DimPlot(data, reduction = "umap", label = T, label.size = 6)
```

## Graph res selection
```{r}
DimPlot(data, reduction = "umap", group.by = c("shortname", "SCT_snn_res.0.8"))
```


# Look at Clusters
Count of cells in each cluster per condition
```{r}
Idents(data) <- "SCT_snn_res.0.8"
n_cells <- FetchData(data, vars = c("ident", "shortname")) %>% 
  dplyr::count(ident, shortname) %>% 
  tidyr::spread(ident, n)

n_cells
```

## Look through gene names
```{r}
gene_list <- Features(data)
gene_list <- str_split(gene_list, "/t")
```

## Cell cycle phase
tutorial: https://hbctraining.github.io/scRNA-seq_online/lessons/06_SC_SCT_normalization.html
** come back to this
```{r}
# DimPlot(data, label = T, split.by = "Phase")
```

# Clustering Analysis
tutorial: https://hbctraining.github.io/scRNA-seq/lessons/08_SC_clustering_quality_control.html
cell type markers: https://docs.google.com/spreadsheets/d/1WksC1b-KfHl0X1AVh4jKY-fs6LN3IR42KIqhN_kpta0/edit#gid=0

Reference paper, SCI single cell popultions: https://doi.org/10.1084/jem.20210040
  Milich et al

Cell types we expect: 
general immune cells 
specialized neural cells (microglia, astrocytes, glia, neurons)
stromal cells 

```{r}
# Select the RNA counts slot to be the default assay
DefaultAssay(data) <- "RNA"

# Normalize RNA data for visualization purposes
data <- NormalizeData(data, verbose = FALSE)
```

# Immune Cell ID
From OneNote 'Gene Marker List' Spine: 
Cd3e, Cd4, Cd8a (T Cells)
Ncr1, Klrb1c, Klre1 (NK Cells)
Cd79a, Cd19, Ms4a1, Iglc3 (B Cells)
Ighm (Plasma) 
Itgam (Cd11b, many cells)
Cd33 (myeloid)
Ly6c2, Cd14, Ccr2 (Monocytes)
Smox, Apoe, Adgre1 (Macrophages)
Tmem119, Trem2, P2ry12 (Microglia)
Camp, Retnlg, Ly6g (Neutrophils)
Cd86, Siglech, Klk1 (DCs)
Hcrtr2, Fcer1a, Il3ra (Basophils)
Ccr3, Siglecf, Cd44 (Eosinophils) 

```{r immune_id, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Cd3e", "Cd4", "Cd8a",
                            "Ncr1", "Klrb1c", "Klre1",
                            "Cd79a", "Cd19", "Ms4a1", "Iglc3",
                            "Ighm",
                            "Itgam",
                            "Cd33",
                            "Ly6c2", "Cd14", "Ccr2",
                            "Smox", "Apoe", "Adgre1",
                            "Tmem119", "Trem2", "P2ry12",
                            "Camp", "Retnlg", "Ly6g",
                            "Cd86", "Siglech", "Klk1",
                            "Hcrtr2", "Fcer1a", "Il3ra",
                            "Ccr3", "Siglecf", "Cd44")) + RotatedAxis()
```

check feature plot, look for myeloid cluster (Cd33+), immune (Cd45+), stromal (Dcn, Serping1), 
epithelial (Qsox1, Mfge3, Epcam, Cdh1), mesenchyman (Vim), Endothelial (Cav1, Cd34, Cd93), lymphatic endothelial (Lyve1, Pdpn),
Fibroblast (Col1a1, Col3a1, Col5a1, Cald1), Eosinophil (Ccr3, Adgre1, Itgam/CD11b, Siglecf, Pgp1), Basophil (Hcrtr2, Fcer1a, Il3ra),
DC (Cd86)
```{r type_id, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c( "Cd33", "Ptprc", 
                                  "Dcn", "Serping1", 
                                  "Osox1", "Mfge3", "Krt5", "Epcam", "Cdh1", 
                                  "Vim", 
                                  "Cav1", "Cd34", "Cd93", 
                                  "Lyve1", "Pdpn",
                                  "Col1a1", "Col3a1", "Col5a1", "Cald1", 
                                  "Ccr3", "Itgam", "Siglecf", "Pgp1", 
                                  "Hcrtr2", "Fcer1a", "Il3ra", 
                                  "Cd86")) + RotatedAxis()
```
Sophia's Markers 
paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9643644/
Single-cell RNA seq identifies anti-cancer immune phenotypes 
from the supplmental file 
Emr1 = Adgre1
Ly6c2 not found- monocyte marker, replaced with Cd14

Mrc1 = Cd206 ? 
Ptprc=cd45, immune; itgam = cd11b, myeloid 
Cd3e = T cells; S100a4 = stromal; Fn1 = HSCs; 
Sftpc = pneumocytes; Limch1 = pneumocytes
Klrb1c = NK; Ly6g / REtnlg = neutrophils
Ly6c2 - monocytes
Emr1 (adgre1). cd68 = macrophages 
Cd34 = HSCs (hematopoietic stem cells)
Elane = granulocytes
Pecam1 = Endothelial; Siglech = DCs
Ms4a1 = B cells 
alv mac = Mrc1, Ear2 (Nr2f6)

```{r type_id_sophia, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Ptprc", "Itgam",
                          "Cd3e", "S100a4", "Fn1", "Sftpc", 
                            "Limch1", "Klrb1c", "Ly6g",
                            "Retnlg", "Cd14", "Cd68", 
                            "Adgre1", "Cd34", "Elane", 
                           "Pecam1", "Siglech", "Ms4a1", "Mrc1", "Nr2f6"
  
)) + RotatedAxis()
```


## transcription factors 
Lineage Markers
https://doi.org/10.1186/s13059-018-1416-2
hematopoietic cells, where TFs such as Gata1, Tal1, Runx1, and Klf1 were specifically active; neuronal cells, which specifically activate TFs such as Sox2, Sox21, and Pax6; epithelial cells, exhibiting high expression of genes that are crucial for epithelial cells, such as Epcam, Cldn6, Cldn7, and Cdh1; and mesoderm-derived cells, expressing marker genes including Col3a1, Pcolce, and Cdh11. 

```{r type_id_TF, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Gata1", "Tal1", "Runx1", "Klf1", 
                            "Sox2", "Sox21", "Pax6",
                            "Epcam", "Cldn6", "Cdh1", 
                            "Col3a1", "Pcolce", "Cdh11"
  
)) + RotatedAxis()
```


# Stromal Cell ID

# Milich 2021 / PanglaoDB
```{r set clusters}
Idents(data) <- data@meta.data$SCT_snn_res.0.8
```

***6/4 try looking at DCs in lung only samples, and in NP+ only sample. Maybe other cell types are blowing out expression
CD11c = Itgax 
```{r DCs, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Itgax", "Cd86", "Siglech", "Klk1", "Cd40", "Cd80")) + RotatedAxis()

```

```{r DCs, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Itgax", "Cd83", "Lamp3", "Cd86",  
                           "Axl", "Dab2", "Fcer1a", "Il6", 
                           "Cd80", "Batf3", "Siglecf")) + RotatedAxis()

```


```{r neutrophils, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("S100a9", "Mmp9", "Ly6g", "Cd177", "Ltf")) + RotatedAxis()
```

monocytes PanglaoDB
Cd14, Lyz, Cfp, Ccr2, Csf3r, Cd7, Tet2, Cd40, Dysf, Cmklr1, 
```{r monocytes, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Cd14","Lyz", "Ccr2", "F10", "Plac8", "Thbs1")) + RotatedAxis()
```

```{r macrophages, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Adgre1", "Cd68", "Itgam", "Smox", "Apoe")) + RotatedAxis()
```


```{r dividing myeloid, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Top2a", "Mki67", "Cdk1", "Ccnb2")) + RotatedAxis()
```

```{r fibroblast, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Col1a1", "Col6a1", "Dcn", "Lum", "Postn")) + RotatedAxis()
```

```{r endothelial, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Pecam1", "Kdr", "Flt1")) + RotatedAxis()
```

### cancer 
```{r 4T1, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Epcam", "Krt18", "Mki67", "Esr1"))
```

### Lung specific cells 
```{r pulmonary_alv_type1, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default' }
DotPlot(data, features = c("Col4a3", "Cldn18", "Fstl3", "Sema3b", "Vegfa" ,"Ager", "Col4a4")) + RotatedAxis()

```


```{r pulmonary_alv_type2, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default' }
DotPlot(data, features = c("Sftpc", "Slc34a2", "Sftpa1", "Nkx2-1", "Etv5" ,"Lamp5", "Muc1")) + RotatedAxis()

```

```{r ionocytes, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Foxi1", "Ctfr", "Ascl3", "Foxn4", "Cyp2f2" ,"Scgb1a1", "Tfcp2l1")) + RotatedAxis()
```

```{r alv_macrophages, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Marco", "Itgax", "Mrc1", "Siglecf", "Ccl3" ,"Car4", "Lyz1")) + RotatedAxis()
```

```{r airway_goblet, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Ggh", "Muc16", "Dusp4", "Muc5b", "Ltf" ,"Muc4", "Muc20")) + RotatedAxis()
```

```{r airway_epithelial, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Aqp1", "Cdh1", "Sec14l3", "Adh7")) + RotatedAxis()
```

### lineage
```{r lineage, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Ptprc", "Itgam", "Cd33", "Cd34", "Pecam1" ,"Epcam", "Col1a1")) + RotatedAxis()
```



## Identifying the Unknown Clusters
```{r join layers set id }
data <- JoinLayers(data)
Idents(data) <- data@meta.data$SCT_snn_res.0.8
```


```{r cluster 3}
FindMarkers(data, ident.1 = 3, only.pos = TRUE)[1:10,]
```

```{r cluster 5}
FindMarkers(data, ident.1 = 5, only.pos = TRUE)[1:10,]
```
```{r cluser 7}
# figure out if these are macrophages or DCs
FindMarkers(data, ident.1 = 7, only.pos = TRUE)[1:10,]
```

```{r cluster 14}
FindMarkers(data, ident.1 = 14, only.pos = TRUE)[1:10,]
```
```{r cluster 15}
FindMarkers(data, ident.1 = 15, only.pos = TRUE)[1:10,]
```
```{r cluster 18}
FindMarkers(data, ident.1 = 18, only.pos = TRUE)[1:10,]
```
```{r cluster 21}
FindMarkers(data, ident.1 = 21, only.pos = TRUE)[1:10,]
```
```{r cluster 24}
FindMarkers(data, ident.1 = 24, only.pos = TRUE)[1:10,]
```
```{r cluster 29}
FindMarkers(data, ident.1 = 29, only.pos = TRUE)[1:10,]
```
```{r cluster 30}
FindMarkers(data, ident.1 = 30, only.pos = TRUE)[1:10,]
```
```{r cluster 31}
FindMarkers(data, ident.1 = 31, only.pos = TRUE)[1:10,]
```
```{r cluster328}
FindMarkers(data, ident.1 = 32, only.pos = TRUE)[1:10,]
```
```{r cluster 33}
FindMarkers(data, ident.1 = 33, only.pos = TRUE)[1:10,]
```
```{r cluster 34}
FindMarkers(data, ident.1 = 34, only.pos = TRUE)[1:10,]
```


## Feature Plot code
```{r pDC feat plot}
FeaturePlot(data, 
            reduction = "umap", 
            features = c("Il3ra", "Gzmb", "Serpinf1", "Itm2c"), 
            min.cutoff = 'q10', 
            label = TRUE)
```

```{r pDCs dot plot, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default'}
DotPlot(data, features = c("Il3ra", "Gzmb", "Serpinf1", "Itm2c")) + 
  RotatedAxis()
```


## Naming Clusters
```{r phenotype}
Idents(data)  <- data@meta.data$SCT_snn_res.0.8
data <- RenameIdents(data, '0' = 'Myeloid', '1' = 'Monocyte', '2' = 'T cell', '3' =
                        'Unknown 1', '4' = 'B cell', '5' = 'Neutrophil',
                      '6' = 'Neutrophil', '7' = 'Mac/DC', '8' = 'CD4 T cell', '9' = 'Epithelial',
                      '10' = 'Unknown 2', '11' = 'CD 4 T cell', '12' = 'B cell', '13' = 'CD8 T cell', 
                      '14' = 'Neutrophil', '15' = 'Endothelial', '16' = 'Endothelial', '17' = 'Neutrophil', 
                      '18' = 'Macrophage', '19' = 'Macrophage', '20' = 'CD4 T cell', '21' = 'Neutrophil',
                      '22' = 'Neutrophil', '23' = 'Macrophage', '24' = 'Endothelial', '25' = 'B cell',
                      '26' = 'Unknown 3', '27' = 'Fibroblast', '28' = '4T1', '29' = 'Metabolic',
                      '30' = 'Mac/DC', '31' = 'DC', '32' = 'B cell', '33' = 'Dividing', 
                      '34' = 'Unknown 4', '35' = 'Myeloid dividing')
data@meta.data$phenotype = data@active.ident

```

```{r}
DimPlot(data, label = TRUE)
```


## Name Lineage Clusters
```{r lineage}
Idents(data)  <- data@meta.data$SCT_snn_res.0.8
data <- RenameIdents(data, '0' = 'Myeloid', '1' = 'Myeloid', '2' = 'Immune', '3' =
                        'Unknown', '4' = 'Immune', '5' = 'Myeloid',
                      '6' = 'Myeloid', '7' = 'Myeloid', '8' = 'Immune', '9' = 'Stromal',
                      '10' = 'Immune', '11' = 'Immune', '12' = 'Immune', '13' = 'Immune', 
                      '14' = 'Myeloid', '15' = 'Endothelial', '16' = 'Endothelial', '17' = 'Myeloid', 
                      '18' = 'Myeloid', '19' = 'Myeloid', '20' = 'Immune', '21' = 'Myeloid',
                      '22' = 'Myeloid', '23' = 'Myeloid', '24' = 'Endothelial', '25' = 'Immune',
                      '26' = 'Unknown', '27' = 'Stromal', '28' = 'Stromal', '29' = 'Unknown',
                      '30' = 'Myeloid', '31' = 'Myeloid', '32' = 'Immune', '33' = 'Immune', 
                      '34' = 'Unknown', '35' = 'Myeloid')
data@meta.data$lineage = data@active.ident
```

```{r}
DimPlot(data, label = TRUE)

```
```{r}
DimPlot(data, reduction = "umap", label = T, group.by = c("shortname", "lineage"))
```


```{r}
DimPlot(data, reduction = "umap", label = TRUE) #+ NoLegend()
```

# Sub-Clustering
## DCs
```{r}

```

```{r cDC, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default' }
DotPlot(data, features = c("Fcer1a", "Cst3")) + RotatedAxis()
```

```{r pDC, fig.height=12, fig.width=15, fig.asp=0.5 , fig.align='default' }
DotPlot(data, features = c("Il3ra", "Gzmb", "Serpinf1", "Itm2c")) + RotatedAxis()
```


# Conserved Marker ID
https://hbctraining.github.io/scRNA-seq/lessons/09_merged_SC_marker_identification.html

# SingleR
some references/tutorials: 
Bioconductor vignette: https://bioconductor.org/packages/release/bioc/vignettes/SingleR/inst/doc/SingleR.html#2_Using_built-in_references
Bioconductor SingleR book: https://bioconductor.org/books/release/SingleRBook/introduction.html#quick-start

Cell References 
celldex: pokedex for celltypes, https://bioconductor.org/packages/3.19/data/experiment/vignettes/celldex/inst/doc/userguide.html


Install packages if not already installed: 
issue with alabaster.base , try alternate install for that package? 
https://www.bioconductor.org/packages/release/bioc/html/alabaster.base.html 
* 5/7/24 KG

```{r}
# BiocManager::install("SingleR")
# BiocManager::install("scRNAseq")
# BiocManager::install("celldex") # this one failed "ERROR: dependency ‘alabaster.base’ is not available for package # ‘celldex’
```

```{r echo=FALSE, results="hide", message=FALSE}
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
library(BiocStyle)
# library(celldex)
library(scRNAseq)
library(SingleR)

```

Search function is throwing an error- package didn't install correctly?? 5/7/24 KG
```{r load ref dataset}
# Find all mm10 datasets involving pancreas or neurons.
scRNAseq::searchDatasets(
   defineTextQuery("GRCm38", field="genome") &
   (defineTextQuery("neuro%", partial=TRUE) | 
    defineTextQuery("spin%", partial=TRUE))
)[,c("name", "title")]
```

## obtain reference dataset

Use scRNAseq package to obtain a reference dataset: 
Zeisel A et al. (2018). Molecular architecture of the mouse nervous system. Cell 174(4), 999-1014.
mouse nervous system single cell RNA-seq from Zeisel et al 2018
Row data contains the gene symbol as well as some relevant per-gene statistics, e.g., the squared coefficient of variance, mean, and whether it was selected for downstream analyses.
Column data contains a wide variety of fields including patient-level information, sample-level sequencing statistics and many flavors of cell type classification. Note that many numeric columns may have NA values if they could not be successfully parsed form the source file.
If location=TRUE, the coordinates of the Ensembl gene models are stored in the rowRanges of the output.
All data are downloaded from ExperimentHub and cached for local re-use. Specific resources can be retrieved by searching for scRNAseq/zeisel-nervous.
```{r scRNAseq ref dataset}
ref_data <- ZeiselNervousData(location = TRUE)
```

Using celldex to obtain a reference dataset: 
```{r celldex ref dataset}
# celldex::searchReferences(
#     defineTextQuery("immun%", partial=TRUE) & 
#     defineTextQuery("10090", field="taxonomy_id")
# )

ms_ref <- celldex::MouseRNAseqData(ensembl = FALSE, cell.ont = "all")

# Immunological Genome Project (ImmGen) 
# microarray of pure mouse immune cells 
immgen_ref <- celldex::ImmGenData(ensembl = FALSE, cell.ont = "all")
```


### singleR using built in reference (Bioconductor tutorial)
```{r}
hpca.se <- HumanPrimaryCellAtlasData()
hpca.se
```

```{r}
hESCs <- LaMannoBrainData('human-es')
hESCs <- hESCs[,1:100]
```

```{r}
pred.hesc <- SingleR(test = data, ref = ref_data, assay.type.test=1,
    labels = ref_data$label.main)
```


```{r}
pred.hesc
# Summarizing the distribution:
table(pred.hesc$labels)
```

### singleR using single-cell references (Bioconductor tutorial)

```{r}
library(scRNAseq)
sceM <- MuraroPancreasData()

# One should normally do cell-based quality control at this point, but for
# brevity's sake, we will just remove the unlabelled libraries here.
sceM <- sceM[,!is.na(sceM$label)]

# SingleR() expects reference datasets to be normalized and log-transformed.
library(scuttle)
sceM <- logNormCounts(sceM)
```

```{r}
sceG <- GrunPancreasData()
sceG <- sceG[,colSums(counts(sceG)) > 0] # Remove libraries with no counts.
sceG <- logNormCounts(sceG) 
```

```{r}
pred.grun <- SingleR(test=sceG, ref=sceM, labels=sceM$label, de.method="wilcox")
table(pred.grun$labels)
```

### Visualization of singleR predictions (Bioconductor tutorial)

```{r}
plotScoreHeatmap(pred.grun)
```

```{r}
plotDeltaDistribution(pred.grun, ncol = 3)

```

```{r}
summary(is.na(pred.grun$pruned.labels))

```

```{r}
all.markers <- metadata(pred.grun)$de.genes
sceG$labels <- pred.grun$labels

# Beta cell-related markers
library(scater)
plotHeatmap(sceG, order_columns_by="labels",
    features=unique(unlist(all.markers$beta))) 
```

### accessing new references
ArrayExpress and GEOquery can be used to access any bulk or single cell datasets in ArrayExpress or GEO


# SessionInfo
```{r}
sessionInfo()
```


